File: CS 471/Program2/program2.pl
#Joseph Camacho-Terrazas
#09/6/2020
#Input: None
#Output: Result of each test
#Preconditions: None
#Postconditions: Print the results from tests and a confirmation message if 2nd condition is reached in comparison statements.

#!/usr/bin/perl

use warnings;

#Create function Evaluate that prints a confirmation message upon evaluation
#Returns 1 for easy comparisons
sub Evaluate {
	printf "Condition has been evaluated \n";
	return 1;
}

#Declare variable a for comparisons
$a = 1;

#Testing "AND" logical operators
printf "===Testing AND with function as second condition===\n";
#Test F and T
if ($a == 0 && Evaluate() == 1) {
	printf "True\n";
} else {
	printf "False\n"
}

printf "\n";

#Test T and F
if ($a == 1 && Evaluate() == 0) {
	printf "True\n";
} else {
	printf "False\n";
}

printf "\n";

#Test T and T
if ($a == 1 && Evaluate() == 1) {
	printf "True\n";
} else {
	printf "False\n";
}

printf "\n";

#Test F and F
if ($a == 0 && Evaluate() == 0) {
	printf "True\n";
} else {
	printf "False\n";
}

printf "\n";

#Testing with function as the first condition just to verify that the function does work properly

printf "===Testing AND with function as first condition===\n";
#Test F and T
if (Evaluate() == 0 && $a == 1) {
	printf "True\n";
} else {
	printf "False\n"
}

printf "\n";

#Test T and F
if (Evaluate() == 1 && $a == 0) {
	printf "True\n";
} else {
	printf "False\n";
}

printf "\n";

#Test T and T
if (Evaluate() == 1 && $a == 1) {
	printf "True\n";
} else {
	printf "False\n";
}

printf "\n";

#Test F and F
if (Evaluate() == 0 && $a == 0) {
	printf "True\n";
} else {
	printf "False\n";
}

File: CS 471/Program2/program2.php
<?php  
//Joseph Camacho-Terrazas
//9/6/2020
//Input: None
//Output: Results of each test
//Preconditions: None
//Postconditions: Print the results from tests and a confirmation message if 2nd condition is reached in comparison statements.

//Create Evaluate function to print a confirmation message if it's evaluated.
//Returns true for easy comparisons
function Evaluate(){
	echo "Condition has been evaluated" .PHP_EOL;
	return True;
}

//Testing "AND" operators
echo "===Testing AND with function as second condition===" .PHP_EOL;
//Test F and T
if (False && Evaluate()) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;

//Test T and F
if (True && Evaluate() == False) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;

//Test T and T
if (True && Evaluate()) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;
	
//Test F and F
if (False && Evaluate() == False) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;

//Testing with function as the first condition just to verify that the function does work properly

echo "===Testing AND with function as first condition===" .PHP_EOL;
//Test F and T
if (Evaluate() == False && True) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;

//Test T and F
if (Evaluate() && False) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;

//Test T and T
if (Evaluate() && True) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;
	
//Test F and F
if (Evaluate() && False) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}
?> 